sklego/meta/ordinal_classification.py

```diff
@@ -0,0 +1,185 @@
+
+        X, y = check_X_y(X, y, estimator=self)
+
+        self.classes_ = np.sort(np.unique(y))
+        self.n_features_in_ = X.shape[1]
+
+        if self.n_jobs is None or self.n_jobs == 1:
+            self.estimators_ = {y_label: self._fit_binary_estimator(X, y, y_label) for y_label in self.classes_[:-1]}
+        else:
+            self.estimators_ = dict(
+                zip(
+                    self.classes_[:-1],
+                    Parallel(n_jobs=self.n_jobs)(
+                        delayed(self._fit_binary_estimator)(X, y, y_label) for y_label in self.classes_[:-1]
+                    ),
+                )
+            )
+        return self
+
+    def predict_proba(self, X):
+        """Predict class probabilities for samples in `X`. The class probabilities of a sample are computed as the
+        difference between the probability of the sample to be in class `y_label` and the probability of the sample to
+        be in class `y_label - 1`.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The data to predict.
+
+        Returns
+        -------
+        array-like of shape (n_samples, n_classes)
+            The predicted class probabilities.
+
+        Raises
+        ------
+        ValueError
+            If `X` has a different number of features than the one seen during `fit`.
+        """
+        check_is_fitted(self, ["estimators_", "classes_"])
+
+        if X.shape[1] != self.n_features_in_:
+            raise ValueError(f"X has {X.shape[1]} features, expected {self.n_features_in_} features.")
+
+        raw_proba = np.array([estimator.predict_proba(X)[:, 1] for estimator in self.estimators_.values()]).T
+        p_y_le = np.column_stack((np.zeros(X.shape[0]), raw_proba, np.ones(X.shape[0])))
+
+        # Equivalent to (p_y_le[:, 1:] - p_y_le[:, :-1])
+        return np.diff(p_y_le, n=1, axis=1)
+
+    def predict(self, X):
+        """Predict class labels for samples in `X` as the class with the highest probability."""
+        return self.classes_[np.argmax(self.predict_proba(X), axis=1)]
+
+    def _fit_binary_estimator(self, X, y, y_label):
+        """Utility method to fit a binary classifier on the dataset where `y <= y_label`.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features )
+            The training data.
+        y : array-like of shape (n_samples,)
+            The target values.
+        y_label : int
+            The label of the class to predict.
+
+        Returns
+        -------
+        fitted_model : scikit-learn compatible classifier
+            The fitted binary classifier.
+        """
+        y_bin = (y <= y_label).astype(int)
+        fitted_model = clone(self.estimator).fit(X, y_bin)
+        if self.use_calibration:
+            return CalibratedClassifierCV(fitted_model, cv="prefit").fit(X, y_bin)
```

Comment:
This is something I would rather have, but open to discuss if actually needed (and eventually which default to have) or only suggested as good practice for the user. I added a _warning_ section in the class docstring to explain why

Reply:
Wouldn't the sklearn API prefer this:

```
pipeline = CalibratedClassifierCV(OrdinalClassifier(estimator))
```

It feels more inline with the API to have users do this themselves, no?